pytorch/containers/hp_tuning/task.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the \"License\");
+# you may not use this file except in compliance with the License.\n",
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an \"AS IS\" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import hypertune
+import torch
+import torch.optim as optim
+import torch.nn as nn
+
+import data_utils
+import model
+
+
+def train(net, train_loader, optimizer):
+    """Create the training loop"""
+    net.train()
+    criterion = nn.BCELoss()
+
+    for batch_index, data in enumerate(train_loader):
+        features = data['features']
+        target = data['target']
+
+        # zero the parameter gradients
+        optimizer.zero_grad()
+        # forward + backward + optimize
+        outputs = net(features)
+        loss = criterion(outputs, target)
+        loss.backward()
+        optimizer.step()
+
+
+def set_value_(item):
+    """Take the predictions and assign them a binary target value."""
+    if item < 0.5:
+        return 0.0
+    else:
+        return 1.0
+
+
+def test(net, test_loader):
+    """Test the DNN"""
+    net.eval()
+    criterion = nn.BCELoss()
```

Comment:
nit: Can we add some details on the implementation of nn.BCEloss(). Maybe a link/paper